events.go

```diff
@@ -144,21 +148,17 @@ func (eo *eventObserver) Change(data []byte) error {
 	eo.lock.RLock()
 	defer eo.lock.RUnlock()
 
-	if eo.listeners == nil {
-		return nil
-	}
-
-	for _, listener := range eo.listeners[e.ID] {
-		//If the listeners do a lot of work ,kicking these off in goroutines might be worth
-		listener(e)
-	}
-
-	if eo.globalListeners == nil {
-		return nil
+	if eo.listeners != nil {
```

Comment:
this nil check is not required, as range does that for you... I see the problem is that if you have no listeners but you have global listeners the global listeners are never "serviced"
so I suggest a simple flow like this:
```
for _, listener := range eo.listeners {
...
}

for _, globalListener := range eo.globalListeners {
...
}

return nil
```

Reply:
Gotcha! TIL.

Maybe I'll be able to pass to go part of the backend interview soon!

Reply:
The first nil check is still needed because you're iterating over eo.listeners[someKey] rather than eo.listeners. Right?

Reply:
I'm sure range should be smart enough to not crash... but just to make sure lets make and example
https://play.golang.org/p/dXNJmnaIhJU

Reply:
GO's smarter than me.

Reply:
Python would've thrown at least 3 errors in this context.